radixdlt-java/src/main/java/com/radixdlt/client/application/translate/validators/RegisterValidatorActionMapper.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the “Software”),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.radixdlt.client.application.translate.validators;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.radixdlt.client.application.translate.ShardedParticleStateId;
+import com.radixdlt.client.application.translate.StageActionException;
+import com.radixdlt.client.application.translate.StatefulActionToParticleGroupsMapper;
+import com.radixdlt.client.atommodel.validators.RegisteredValidatorParticle;
+import com.radixdlt.client.atommodel.validators.UnregisteredValidatorParticle;
+import com.radixdlt.client.core.atoms.ParticleGroup;
+import com.radixdlt.client.core.atoms.particles.Particle;
+import com.radixdlt.client.core.atoms.particles.SpunParticle;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+/**
+ * Action mapper for {@link RegisterValidatorAction}s, implemented using {@link ValidatorRegistrationState}.
+ */
+public class RegisterValidatorActionMapper implements StatefulActionToParticleGroupsMapper<RegisterValidatorAction> {
+	@Override
+	public Set<ShardedParticleStateId> requiredState(RegisterValidatorAction action) {
+		return ImmutableSet.of(
+			ShardedParticleStateId.of(RegisteredValidatorParticle.class, action.getValidator()),
+			ShardedParticleStateId.of(UnregisteredValidatorParticle.class, action.getValidator())

```

Comment:
I would think you only need UnregisteredValidatorParticle here no?

Reply:
Leaving for now as discussed